### PR TITLE
fix: fix infinite recursion during some redirects

### PR DESCRIPTION
**Pull request by Bkucera**
## PR Checklist:
- [ ] I have run `npm test` locally and all tests are passing.
- [ ] I have added/updated tests for any new behavior.
       
- [ ] If this is a significant change, an issue has already been created where the problem / solution was discussed: [N/A, or add link to issue here]

### CODE REVIEW

File: request.js [hunk 1]
```diff
@@ -991,6 +991,7 @@ Request.prototype.onRequestResponse = function (response) {
     // Hack to emit end on close because of a core bug that never fires end
     response.on('close', function () {
       if (!self._ended) {
+        self._ended = true
```
Comment (flotwig) [inline request.js]: mildly worried that just setting this flag to true will cause issues with stream flow, let me set up tests for this repo and see

Comment (Bkucera) [reply request.js]: ok, I could always set another internal prop and `if` that

Comment (flotwig) [reply request.js]: all the tests pass here & in Cypress, so I suppose this is fine

**Pull request merged**